desktop/SmesherManager.ts

```diff
@@ -38,18 +42,75 @@ class SmesherManager extends AbstractManager {
 
   private genesisID: string;
 
+  private netName: string;
+
   constructor(
     mainWindow: BrowserWindow,
     configFilePath: string,
-    genesisID: string
+    genesisID: string,
+    netName: string | undefined
   ) {
     super(mainWindow);
     this.smesherService = new SmesherService();
     this.smesherService.createService();
     this.configFilePath = configFilePath;
     this.genesisID = genesisID;
+    this.netName = netName || '';
   }
 
+  updateSmeshingOptsFile = async (
+    netName: string,
+    smeshingOpts: Partial<NodeConfig['smeshing']>
+  ) => {
+    const customNodeConfig = await loadOrCreateCustomConfig(netName);
+    let configSmeshing;
+
+    // case of delete smeshing opts
+    if (R.isEmpty(smeshingOpts)) {
+      const clientConfig = await this.loadConfig();
+      const genesisID = generateGenesisIDFromConfig(clientConfig);
+      configSmeshing = safeSmeshingOpts(undefined, genesisID);
+    } else {
+      configSmeshing = {
+        ...customNodeConfig.smeshing,
+        ...smeshingOpts,
+      };
+    }
+
+    const batchSize =
+      configSmeshing?.['smeshing-opts']?.['smeshing-opts-compute-batch-size'];
+    if (
+      batchSize &&
+      batchSize === DEFAULT_SMESHING_BATCH_SIZE &&
+      configSmeshing['smeshing-opts']
+    ) {
+      configSmeshing['smeshing-opts'][
+        'smeshing-opts-compute-batch-size'
+      ] = R.pathOr(
+        DEFAULT_SMESHING_BATCH_SIZE,
+        ['smeshing-opts', 'smeshing-opts-compute-batch-size'],
+        customNodeConfig?.smeshing
+      );
+    }
+
+    await updateSmeshingOptsInCustomConfig(netName, configSmeshing);
+
+    await this.updateConfig(netName, { smeshing: configSmeshing });
```

Comment:
Can't we get rid of `updateSmeshingOptsInCustomConfig` and update the custom config within the `updateConfig` function?

Reply:
And, should not it merge the loaded custom config with the modified smeshing? 🤔 
```suggestion
    await updateSmeshingOptsInCustomConfig(netName, configSmeshing);

    await this.updateConfig(netName, { ...customNodeConfig, smeshing: configSmeshing });
```